src/handlers/spid.ts

```diff
@@ -117,3 +141,159 @@ export const accessLogHandler = (
     }
   );
 };
+
+export const acs: (
+  config: IConfig
+) => // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+AssertionConsumerServiceT = config => async user =>
+  pipe(
+    user,
+    SpidUser.decode,
+    E.mapLeft(errs => toResponseErrorInternal(errorsToError(errs))),
+    // binds the Spid level to the SpidUser
+    E.bindW("authnContextClassRef", spidUser =>
+      pipe(
+        spidUser.getAssertionXml(),
+        t.string.decode,
+        E.chainW(assertion =>
+          pipe(
+            assertion,
+            safeXMLParseFromString,
+            O.chain(getSpidLevelFromSAMLResponse),
+            E.fromOption(() => new Error("Spid level retrieval failed"))
+          )
+        ),
+        E.getOrElse(() => SpidLevelEnum["https://www.spid.gov.it/SpidL2"]),
+        E.of
+      )
+    ),
```

Comment:
Why we are parsing the `SAMLResponse`, the value we need is already parsed from `io-spid-commons` and included into the property `authnContextClassRef` of the user.
Just edit the `SpidUser` type adding the missing property.
```suggestion
```

Reply:
There is no `authnContextClassRef` within the user coming from `spid-commons`. 
In io-backend we also retrieve it from SAML Response (see [here](https://github.com/pagopa/io-backend/blob/f781120073e09a360043e2d78da712497c2d0522/src/types/user.ts#L253-L254)).

It's true that we should refactor spid-commons to return it, but we can postpone to another PR.

Reply:
My bad, the value is not returned from `io-spid-commons`, we need to extract it from the SAMLResponse